mmf/utils/checkpoint.py

```diff
@@ -154,137 +164,153 @@ def load_state_dict(self):
-                            == own_attr.replace(value, "")
-                        ):
-                            self.trainer.writer.write(
-                                "Copying " + attr + " " + own_attr
-                            )
-                            own_state[own_attr].copy_(new_dict[attr])
-            self.trainer.writer.write("Pretrained model loaded")
-
-    def _load_state_dict_mapping(self, ckpt_model):
-        model = self.trainer.model
-        attr_mapping = {
-            "image_feature_encoders": "img_feat_encoders",
-            "image_feature_embeddings_list": "img_embeddings_list",
-            "image_text_multi_modal_combine_layer": "multi_modal_combine_layer",
-            "text_embeddings": "text_embeddings",
-            "classifier": "classifier",
-        }
+        elif "best_iteration" in ckpt:
+            # Preserve old behavior for old checkpoints where we always
+            # load best iteration
+            if ckpt_config.resume_best and "current_iteration" in ckpt:
+                self.trainer.current_iteration = ckpt["current_iteration"]
+            else:
+                self.trainer.current_iteration = ckpt.get(
+                    "best_iteration", self.trainer.current_iteration
+                )
 
-        data_parallel = registry.get("data_parallel")
+            self.trainer.num_updates = self.trainer.current_iteration
 
-        if not data_parallel:
-            for key in attr_mapping:
-                attr_mapping[key.replace("module.", "")] = attr_mapping[key]
-                attr_mapping.pop(key)
+        registry.register("current_iteration", self.trainer.current_iteration)
+        registry.register("num_updates", self.trainer.num_updates)
 
-        for key in attr_mapping:
-            getattr(model, key).load_state_dict(ckpt_model[attr_mapping[key]])
+        self.trainer.current_epoch = ckpt.get("best_epoch", self.trainer.current_epoch)
+        registry.register("current_epoch", self.trainer.current_epoch)
+
+    def _load_pretrained(self, ckpt):
+        model = self.trainer.model
+        own_state = model.state_dict()
+        mapping = self.trainer.config.checkpoint.pretrained_state_mapping
+        for key, value in mapping.items():
+            key += "."
+            value += "."
+            for attr in ckpt:
+                for own_attr in own_state:
+                    formatted_attr = model.format_state_key(attr)
+                    if (
+                        key in formatted_attr
+                        and value in own_attr
+                        and formatted_attr.replace(key, "")
+                        == own_attr.replace(value, "")
+                    ):
+                        self.trainer.writer.write("Copying " + attr + " " + own_attr)
+                        own_state[own_attr].copy_(ckpt[attr])
+        self.trainer.writer.write("Pretrained model loaded")
+
+    def upgrade_state_dict(self, state_dict):
```

Comment:
I think this `upgrade_state_dict` needs to be added in the `load_pretrained_model` method in L34 while loading.

Reply:
Fixed. Thanks.